src/app/navigation/vertical-navigation.component.ts

```diff
@@ -0,0 +1,665 @@
+
+/**
+ * Vertical navigation component
+ */
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'pfng-vertical-navigation',
+  styleUrls: ['./vertical-navigation.component.less'],
+  templateUrl: './vertical-navigation.component.html',
+  host: {
+    '(window:resize)': 'onResize($event)'
+  }
+})
+export class VerticalNavigationComponent implements OnInit, OnDestroy {
+
+  /**
+   * Source for the brand logo at the top
+   */
+  @Input() brandSrc: string;
+
+  /**
+   * Alternate text to display for the logo
+   */
+  @Input() brandAlt: string;
+
+  /**
+   * Boolean to indicate whether or not to show badges
+   */
+  @Input() showBadges: boolean;
+
+  /**
+   * Indicates whether or not to allow the secondary to persist
+   */
+  @Input() persistentSecondary: boolean;
+
+  /**
+   * Allow pinnable menus when they are open
+   */
+  @Input() pinnableMenus: boolean;
+
+  /**
+   * Hide menu items
+   */
+  @Input() hiddenIcons: boolean;
+
+  /**
+   * The navigation items used to build the menu
+   */
+  @Input() items: NavigationItemConfig[];
+
+  /**
+   * Sets an active flag on items when they are selected
+   */
+  @Input() updateActiveItemsOnClick: boolean;
+
+  /**
+   * Indicates whether or not this is a mobile friendly navigation
+   */
+  @Input() ignoreMobile: boolean;
+
+  /**
+   * Hide top banner optionally
+   */
+  @Input() hideTopBanner: boolean;
+
+  /**
+   * This event is fired any time the user has initiated navigation
+   */
+  @Output('onNavigatedEvent') navigatedEvent = new EventEmitter();
+
+  /**
+   * This event is fired any time an item in the navigation is clicked
+   */
+  @Output('onItemClickEvent') itemClickEvent = new EventEmitter();
+
+  private activeSecondary: boolean;
+  private showMobileNav: boolean;
+  private showMobileSecondary: boolean;
+  private showMobileTertiary: boolean;
+  private hoverSecondaryNav: boolean;
+  private hoverTertiaryNav: boolean;
+  private collapsedSecondaryNav: boolean;
+  private collapsedTertiaryNav: boolean;
+  private navCollapsed: boolean;
+  private navHoverTimeout: number;
+  private forceHidden: boolean;
+  private inMobileState: boolean;
+  private routeChangeListener: any;
+
+  // Private internal functions
+  private breakpoints: any = {
+    'tablet': 768,
+    'desktop': 1200
+  };
+  private explicitCollapse: boolean = false;
+  private hoverDelay: number = 500;
+  private hideDelay: number = this.hoverDelay + 200;
+
+  /**
+   * The default constructor
+   */
+  constructor(private elementRef: ElementRef, private renderer: Renderer2, private router: Router) {
+  }
+
+  ngOnInit() {
+    this.activeSecondary = false;
+    this.showMobileNav = false;
+    this.showMobileSecondary = false;
+    this.showMobileTertiary = false;
+    this.hoverSecondaryNav = false;
+    this.hoverTertiaryNav = false;
+    this.collapsedSecondaryNav = false;
+    this.collapsedTertiaryNav = false;
+    this.navCollapsed = false;
+    this.forceHidden = false;
```

Comment:
Can these defaults be declared on lines 83-95? Unless the component is being reinitialized, it might be cleaner.